From d9f6480ee61a57f805065c3f6c759951dbe8256f Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Marc-Andr=C3=A9=20Moreau?= <mamoreau@devolutions.net>
Date: Thu, 2 Nov 2023 16:27:45 -0400
Subject: [PATCH] [PATCH] append version.rc files in vcxproj automatically

---
 src/common.build.post.props | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/src/common.build.post.props b/src/common.build.post.props
index 1326e062b..410f93368 100644
--- a/src/common.build.post.props
+++ b/src/common.build.post.props
@@ -113,4 +113,11 @@
       <ForcedIncludeFiles>$(SolutionDir)\bin\$(Configuration)\inc\TilFeatureStaging.h;%(ForcedIncludeFiles)</ForcedIncludeFiles>
     </ClCompile>
   </ItemDefinitionGroup>
+
+  <PropertyGroup><BuildDependsOn>AppendVersionRc;$(BuildDependsOn)</BuildDependsOn></PropertyGroup>
+  <Target Name="AppendVersionRc" Condition="Exists('$(ProjectDir)\version.rc')">
+    <ItemGroup>
+      <ResourceCompile Condition="'@(ResourceCompile)'==''" Include="version.rc" />
+    </ItemGroup>
+  </Target>
 </Project>
-- 
2.39.1.windows.1

