From 9ec5a422006a854eee65d4dfc7634c364f19d49a Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Marc-Andr=C3=A9=20Moreau?= <marcandre.moreau@gmail.com>
Date: Fri, 25 Nov 2022 15:22:17 -0500
Subject: [PATCH] Add support for WT_BASE_SETTINGS_PATH environment variable

---
 src/cascadia/TerminalSettingsModel/FileUtils.cpp | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/src/cascadia/TerminalSettingsModel/FileUtils.cpp b/src/cascadia/TerminalSettingsModel/FileUtils.cpp
index bc9100afa..1743c80ca 100644
--- a/src/cascadia/TerminalSettingsModel/FileUtils.cpp
+++ b/src/cascadia/TerminalSettingsModel/FileUtils.cpp
@@ -41,6 +41,16 @@ namespace winrt::Microsoft::Terminal::Settings::Model
 
             return parentDirectoryForSettingsFile;
         }();
+
+        static std::filesystem::path baseSettingsPathEnv = []() {
+            return std::filesystem::path{ wil::TryGetEnvironmentVariableW<std::wstring>(L"WT_BASE_SETTINGS_PATH") };
+        }();
+
+        if (!baseSettingsPathEnv.empty())
+        {
+            return baseSettingsPathEnv;
+        }
+
         return baseSettingsPath;
     }
 
-- 
2.36.0.windows.1

