From 970cd1c4a1b91d629bba65ee8d130a33928aae01 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Marc-Andr=C3=A9=20Moreau?= <marcandre.moreau@gmail.com>
Date: Tue, 6 Dec 2022 15:36:10 -0500
Subject: [PATCH] append version.rc files in vcxproj automatically

---
 src/common.build.post.props | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/src/common.build.post.props b/src/common.build.post.props
index 47b2ccd03..17f210b0e 100644
--- a/src/common.build.post.props
+++ b/src/common.build.post.props
@@ -77,4 +77,11 @@
       <ForcedIncludeFiles>$(SolutionDir)\bin\$(Configuration)\inc\TilFeatureStaging.h;%(ForcedIncludeFiles)</ForcedIncludeFiles>
     </ClCompile>
   </ItemDefinitionGroup>
+
+  <PropertyGroup><BuildDependsOn>AppendVersionRc;$(BuildDependsOn)</BuildDependsOn></PropertyGroup>
+  <Target Name="AppendVersionRc" Condition="Exists('$(ProjectDir)\version.rc')">
+    <ItemGroup>
+      <ResourceCompile Condition="'@(ResourceCompile)'==''" Include="version.rc" />
+    </ItemGroup>
+  </Target>
 </Project>
-- 
2.36.0.windows.1

